app/src/main/java/com/navapp/navigation/route/RoutePlanner.java

```diff
@@ -0,0 +1,64 @@
+package com.navapp.navigation.route;
+
+import com.google.maps.DirectionsApi;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PendingResult;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.DirectionsResult;
+import com.google.maps.model.GeocodedWaypoint;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RoutePlanner {
+    private final GeoApiContext context;
+    private final DirectionsApi.RouteRestriction[] restrictions;
+
+    public RoutePlanner(GeoApiContext context) {
+        this(context, new ArrayList<>());
+    }
+
+    public RoutePlanner(GeoApiContext context, List<DirectionsApi.RouteRestriction> restrictions) {
+        this.context = context;
+        DirectionsApi.RouteRestriction[] tmp = new DirectionsApi.RouteRestriction[restrictions.size()];
+        this.restrictions = restrictions.toArray(tmp);
+    }
+
+    public void callbackOrderedRouteIds(String originId, List<String> waypointIds, String destinationId, PendingResult.Callback<List<String>> callback) {
```

Comment:
`@Override`

Reply:
Ummm this class does not extend or implement anything... I can, of course, make an interface out of this method, but I am not sure that this is needed...

Reply:
I must have imagined one....